build/build_system.bzl

```diff
@@ -235,3 +246,101 @@ def git_or_local_repository(name, branch, **kwargs):
             branch = branch,
             **kwargs
         )
+
+def _kong_install_impl(ctx):
+    outputs = []
+    strip_path = ctx.attr.strip_path
+    label_path = ctx.attr.src.label.workspace_name + "/" + ctx.attr.src.label.name
+    if not strip_path:
+        strip_path = label_path
+    prefix = ctx.attr.prefix
+    if prefix:
+        prefix = prefix + "/"
+
+    for file in ctx.files.src:
+        # skip top level directory
+        if file.short_path.endswith(label_path) or file.short_path.endswith(strip_path):
+            continue
+
+        # strip ../ from the path
+        path = "/".join(file.short_path.split("/")[1:])
+
+        # skip foreign_cc generated copy_* targets
+        if path.startswith(ctx.attr.src.label.workspace_name + "/copy_"):
```

Comment:
``` suggestion
        # skip foreign_cc generated copy_* targets
        # `label.workspace_name` has been deprecated in the Bazel v7.1.0,
        # we should replace it with `label.repo_name` after upgrading
        # to the Bazel v7.
        # https://bazel.build/versions/7.1.0/rules/lib/builtins/Label
        if path.startswith(ctx.attr.src.label.workspace_name + "/copy_"):
            if not path.startswith(ctx.attr.src.label.workspace_name + "/copy_" + ctx.attr.src.label.name):
                fail("Expected any path starts with `copy_` was generated by foreign_cc, but `%s` is not look like been generated by it" % path)

            continue    
```

Maybe we can check it by inspecting the type of label's provider (`CcInfo`)? This is more safe but more complex.

Reply:
that's a good idea, let me try

Reply:
It will then be
```starlark
        last_two = "/".join(ctx.attr.src[OutputGroupInfo].gen_dir.to_list()[0].path.split("/")[-2:])
        if ForeignCcDepInfo in ctx.attr.src and path.endswith(last_two)
```
which I feel like is less readable than the comparing `/copy_xxx` approach, how do you think?

Reply:
But I think this approach will make sense in the `kong_template_file` rule when getting the folder of the output. Right now we simply use the shortest path from all outputs. I will update that.

Reply:
> It will then be
> 
> ```python
>         last_two = "/".join(ctx.attr.src[OutputGroupInfo].gen_dir.to_list()[0].path.split("/")[-2:])
>         if ForeignCcDepInfo in ctx.attr.src and path.endswith(last_two)
> ```
> 
> which I feel like is less readable than the comparing `/copy_xxx` approach, how do you think?

Ah, no one can understand what that meant.  How about this?

``` python
        # skip foreign_cc generated copy_* targets
        # `label.workspace_name` has been deprecated in the Bazel v7.1.0,
        # we should replace it with `label.repo_name` after upgrading
        # to the Bazel v7.
        # https://bazel.build/versions/7.1.0/rules/lib/builtins/Label
        if path.startswith(ctx.attr.src.label.workspace_name + "/copy_") and ForeignCcDepInfo in ctx.attr.src:
            if path.startswith(ctx.attr.src.label.workspace_name + "/copy_" + ctx.attr.src.label.name):
                continue
```

Reply:
fixed